sway-core/src/asm_generation/from_ir.rs

```diff
@@ -1134,7 +1134,7 @@ impl<'ir> AsmBuilder<'ir> {
             .map(|idx| format!("{}", idx))
             .collect::<Vec<String>>()
             .join(",");
-        if value_size <= 8 {
+        if field_type.is_copy_type() {
```

Comment:
This change here actually breaks the language test `enum_init_fn_call`.. I'm not sure why yet. I know that the change is necessary and correct because the test I added fails without it.